staging/src/k8s.io/apiserver/pkg/server/width/width.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package width
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// DefaultRequestWidthFinder returns '1' as the width of any
+// given request, this is in keeping with the current behavior.
+var DefaultRequestWidthFinder = &defaultWidthFinder{}
+
+// NewChainedRequestWidthFinder returns a RequestWidthFinder that executes
+// the given chain of RequestWidthFinder(s) one at a time in the specified order
+// and returns the "width" of a given request.
+// It stops when a RequestWidthFinder from the chain has returned the "width"
+// of the given request.
+// If no RequestWidthFinder from the chain returned a "width" for the given request
+// the the chain is exhausted and we fallback to the default finder which returns
+// '1' as the "width" of the given request.
+func NewChainedRequestWidthFinder(chain ...RequestWidthFinder) RequestWidthFinder {
+	return &chainedWidthFinder{
+		fallback: &defaultWidthFinder{},
+		chain:    chain,
+	}
+}
+
+// RequestWidthFinder finds the "width" of the given request.
+type RequestWidthFinder interface {
+	// Name returns the name of the RequestWidthFinder, it can be useful in
+	// logging error, for example if we have a chain of RequestWidthFinder(s)
+	// we can include the name of the RequestWidthFinder in the error log.
+	Name() string
+
+	// FindWidth calculates the "width" of a given request.
```

Comment:
How about EstimateWidth() ? Or ComputeWidth?

[related to the comment above about the name of the interface]